### ISSUE TITLE
R5.4 Rules rearchitecture

### ISSUE THREAD

**Issue by MHoov** [IBMa/equal-access#1654]
Refer to mural R5.4 _add link_

## Epic Details
- **User Story**: Users of the Accessibility Checker tools will have more accurate mappings to WCAG Success criteria
- **Playback Demonstration**: Will be able to show different SC mappings for the same rule based on what reason code was detected. And better ACT mappings 
- **Metric**: ACT score should be improved (snapshot before PR merged)
- **Comms**: Release notes

## What project/section is the Epic for?

- [engine](https://github.com/IBMa/equal-access/issues#workspaces/equal-access-61ae71f9f8b4ec001e136462/board?labels=engine&releases=Z2lkOi8vcmFwdG9yL1JlbGVhc2UvOTQ1MzM)
- Possible CI/CD packages impacts

## Definition of Done

- [ ] Results are added to the Release Mural with a description of what was accomplished, KPIs, etc
- [ ] All content has been reviewed for grammar, spelling, style and technical accuracy
- [ ] All code committed to main branch
- [ ] All unit tests pass (including accessibility)
- [ ] Demo works in staging environment
- [ ] At least one performance metric / monitor constructed
